pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_zone_config_system_tenant failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_zone_config_system_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/2d65f2ff-cb8e-46cd-9b3a-31cc36638b8a?testReportRun=1&testReportShard=39&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on master @ [eae66954f7229c24f31cf2993cd9a28567e82f82](https://github.com/cockroachdb/cockroach/commits/eae66954f7229c24f31cf2993cd9a28567e82f82):


```
=== RUN   TestLogic_zone_config_system_tenant
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogic_zone_config_system_tenant2641446766
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[11:38:30] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 5 statements
[11:39:23] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant with config fakedist-disk: 23 tests, 2 failures
[11:39:30] --- total progress: 23 statements
--- total: 23 tests, 2 failures
    logic.go:4534: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_zone_config_system_tenant2641446766
--- FAIL: TestLogic_zone_config_system_tenant (73.37s)
=== RUN   TestLogic_zone_config_system_tenant/regression_93614
    logic.go:3193: let $t_id = 110
[11:38:37] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 22 statements
    logic.go:3131: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant:120: SELECT crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)
        FROM system.span_configurations
        WHERE end_key > (SELECT crdb_internal.table_span(110)[1])
        expected:
            
        but found (query options: "retry") :
            {"gcPolicy": {"ttlSeconds": 14400}, "numReplicas": 3, "rangeMaxBytes": "67108864", "rangeMinBytes": "1048576"}
            
[11:39:23] --- progress: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 23 statements
    logic.go:2331: 
         /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant:126: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_zone_config_system_tenant/regression_93614 (50.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=39</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_zone_config_system_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51698

--- a/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant
+++ b/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant
@@ -121,6 +121,14 @@ SELECT crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)
 FROM system.span_configurations
 WHERE end_key > (SELECT crdb_internal.table_span($t_id)[1])
 ----
+{"gcPolicy": {"ttlSeconds": 14400}, "numReplicas": 3, "rangeMaxBytes": "67108864", "rangeMinBytes": "1048576"}
+
+# Run the same query again and make sure there is 1 row. This assertion is
+# only here to prevent the previous test case from being rewritten accidentally.
+statement count 1
+SELECT crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)
+FROM system.span_configurations
+WHERE end_key > (SELECT crdb_internal.table_span($t_id)[1])
 
 statement ok
 CREATE TABLE db2.t2 (i INT PRIMARY KEY);

